### PR TITLE
refactor: datepicker component

### PR DESCRIPTION
**Pull request by wildergd**
fix: #1584 

## Changes proposed in this PR:
- refactor DatePicker component 

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/DatePicker/index.js [hunk 1]
```diff
@@ -249,6 +221,8 @@ DatePicker.defaultProps = {
     className: undefined,
     style: undefined,
     locale: undefined,
+    selectionType: 'single',
+    variant: 'single',
 };
 
 export default withReduxForm(DatePicker);
```
Comment (yvmunayev) [inline src/components/DatePicker/index.js:228]: We already made a global hook (useReduxForm) for ReduxForm integration, check in FileSelector

File: src/components/DatePicker/index.js [hunk 2]
```diff
@@ -1,179 +1,147 @@
...
+        newValue => {
+            if (selectionType === 'single') {
+                closeModal();
+                onChange(newValue);
+            } else if (newValue.length > 1) {
+                closeModal();
+                onChange(newValue);
+            } else {
+                onChange(newValue);
+            }
```
Comment (yvmunayev) [inline src/components/DatePicker/index.js:76]: Check here

File: src/components/DatePicker/index.js (hunk 1 above)
Comment (wildergd) [reply src/components/DatePicker/index.js:228]: useReduxForm is causing focusable test to fail

Comment (wildergd) [reply src/components/DatePicker/index.js:228]: i think the jestMatcher for toBeDocusable needs modifications in order to work with functional components created with React.forwardRef

File: src/components/DatePicker/hooks/useFormatDate.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { useMemo } from 'react';
+import formatDate from '../helpers/formatDate';
+import { isEmptyRange } from '../../Calendar/helpers';
+
+export default function useFormatDate(value, format, locale, selectionType) {
```
Comment (LeandroTorresSicilia) [inline src/components/DatePicker/hooks/useFormatDate.js:5]: we should use an object when pass to many arguments like:
```
useFormatDate({ value, format, locale, selectionType })
```

**Pull request merged**